src/packet/signature.js

```diff
@@ -354,9 +373,12 @@ class SignaturePacket {
     const critical = bytes[mypos] & 0x80;
     const type = bytes[mypos] & 0x7F;
 
-    if (!hashed && !allowedUnhashedSubpackets.has(type)) {
-      this.unhashedSubpackets.push(bytes.subarray(mypos, bytes.length));
-      return;
+    if (!hashed) {
+      if (!allowedUnhashedSubpackets.has(type)) {
+        this.unhashedSubpackets.push(bytes.subarray(mypos, bytes.length));
+        return;
+      }
+      this.allowedUnhashedSubpackets.push(bytes.subarray(mypos, bytes.length));
```

Comment:
Tbh, I would just change the condition and store all of them in `unhashedSubpackets`. There's nothing saying that those are only the disallowed ones, it's just that for the others there was no reason (until now) to store them there. It's technically a small breaking change, but imo we change it in a minor release.

```suggestion
      this.unhashedSubpackets.push(bytes.subarray(mypos, bytes.length));
      if (!allowedUnhashedSubpackets.has(type)) {
        return;
      }
```

Reply:
I tried this approach, but it doesn't work, because when we overwrite the `allowedUnhashedSubpackets` we do not intend to rewrite all the `unhashedSubpackets`, but just the one the implementation understands (i.e. the allowed ones).

In other words, there might be some packet in the unhashed section that we do not want to overwrite when re-signing, therefore we need to store them separately or flag them as "to be overwritten" or "not to be overwritten" on re-sign.

Reply:
OK, but.. do we rely on preserving unhashed subpackets when re-signing somewhere?
Re-signing in general isn't really something that's often done in OpenPGP.js; we leave the signatures alone by default (unlike go-crypto).

Reply:
Yes, we had dozens of tests failing by not preserving the unhashed subpackets. I didn't inspect most of them, since I'm not very familiar with the codebase, but it seemed to be a pretty rooted assumption.